projects/igniteui-angular/src/lib/query-builder/query-builder.component.ts

```diff
@@ -911,12 +912,15 @@ export class IgxQueryBuilderComponent implements AfterViewInit, OnDestroy {
                     DEFAULT_PIPE_DATE_TIME_FORMAT : DEFAULT_PIPE_DATE_FORMAT;
         }
 
-        if (!field.defaultDateTimeFormat) {
-            field.defaultDateTimeFormat = DEFAULT_DATE_TIME_FORMAT;
-        }
+        this.setDateTimeEditorFormat(field, field.pipeArgs.format);
+    }
 
-        if (!field.defaultTimeFormat) {
-            field.defaultTimeFormat = DEFAULT_TIME_FORMAT;
+    private setDateTimeEditorFormat(field: FieldType, format: string) {
+        if (!DateTimeUtil.isDateTimeDataType(field.dataType)) {
+            return;
+        }
+        if (!field.editorOptions?.dateTimeFormat) {
+            field.editorOptions.dateTimeFormat = format;
```

Comment:
Oh, also not a fan of modifying these field options if those are directly the user-provided ones, but that seems to be a general issue with the QB handling looking around